host/src/attribute/mod.rs

```diff
@@ -1,216 +1,68 @@
+mod consts;
+mod data;
+pub mod server;
+pub use consts::*;
+pub use data::*;
+use heapless::Vec;
+
 use core::cell::RefCell;
 use core::fmt;
+use core::ops::ControlFlow;
 
 use embassy_sync::blocking_mutex::raw::RawMutex;
 use embassy_sync::blocking_mutex::Mutex;
 
 use crate::att::AttErrorCode;
-use crate::cursor::WriteCursor;
 pub use crate::types::uuid::Uuid;
 use crate::Error;
 
-pub const GENERIC_ACCESS_SERVICE_UUID16: Uuid = Uuid::Uuid16(0x1800u16.to_le_bytes());
-pub const CHARACTERISTIC_DEVICE_NAME_UUID16: Uuid = Uuid::Uuid16(0x2A00u16.to_le_bytes());
-pub const CHARACTERISTIC_APPEARANCE_UUID16: Uuid = Uuid::Uuid16(0x2A03u16.to_le_bytes());
-
-pub const GENERIC_ATTRIBUTE_SERVICE_UUID16: Uuid = Uuid::Uuid16(0x1801u16.to_le_bytes());
-
-pub const PRIMARY_SERVICE_UUID16: Uuid = Uuid::Uuid16(0x2800u16.to_le_bytes());
-pub const SECONDARY_SERVICE_UUID16: Uuid = Uuid::Uuid16(0x2801u16.to_le_bytes());
-pub const INCLUDE_SERVICE_UUID16: Uuid = Uuid::Uuid16(0x2802u16.to_le_bytes());
-pub const CHARACTERISTIC_UUID16: Uuid = Uuid::Uuid16(0x2803u16.to_le_bytes());
-pub const CHARACTERISTIC_CCCD_UUID16: Uuid = Uuid::Uuid16(0x2902u16.to_le_bytes());
-pub const GENERIC_ATTRIBUTE_UUID16: Uuid = Uuid::Uuid16(0x1801u16.to_le_bytes());
-
-#[derive(Debug, Clone, Copy)]
+#[derive(Debug, Clone, Copy, PartialEq, Eq)]
 #[repr(u8)]
+/// An enum of possible characteristic properties
+///
+/// Ref: BLUETOOTH CORE SPECIFICATION Version 6.0, Vol 3, Part G, Section 3.3.1.1 Characteristic Properties
 pub enum CharacteristicProp {
+    /// Permit broadcast of the Characteristic Value
+    ///
+    /// If set, permits broadcasts of the Characteristic Value using Server Characteristic
+    /// Configuration Descriptor.
     Broadcast = 0x01,
+    /// Permit read of the Characteristic Value
     Read = 0x02,
+    /// Permit writes to the Characteristic Value without response
     WriteWithoutResponse = 0x04,
+    /// Permit writes to the Characteristic Value
     Write = 0x08,
+    /// Permit notification of a Characteristic Value without acknowledgment
     Notify = 0x10,
+    /// Permit indication of a Characteristic Value with acknowledgment
     Indicate = 0x20,
+    /// Permit signed writes to the Characteristic Value
     AuthenticatedWrite = 0x40,
+    /// Permit writes to the Characteristic Value without response
     Extended = 0x80,
 }
 
-pub struct Attribute<'a> {
+#[derive(PartialEq, Eq)]
+pub struct Attribute<'d> {
+    /// Attribute type UUID
+    ///
+    /// Do not mistake it with Characteristic UUID
```

Comment:
I'm not sure if this is entirely correct, every characteristic has a UUID, but there are different types, like the characteristic declaration or the characteristic value. This UUID can be either.